cedar-policy-core/src/parser/err.rs

```diff
@@ -47,9 +54,288 @@ pub enum ParseError {
+    #[error("internal invariant violated. No parse errors were reported but annotation information was missing")]
+    AnnotationInvariantViolation,
+    /// Returned for an invalid use of `{}`
+    #[error("invalid use of `{{}}`")]
+    InvalidBraces,
+    /// Returned for a non-parse-able string literal
+    #[error("invalid string literal: `\"{0}\"`")]
+    InvalidString(String),
+    /// Returned for attempting to use an arbitrary variable name. Cedar does not support arbitrary variables.
+    #[error("arbitrary variables are not supported; did you mean to enclose `{0}` in quotes to make a string?")]
+    ArbitraryVariable(crate::ast::Name),
+    /// Returned for attempting to use an invalid attribute name
+    #[error(
+        "invalid attribute name. Attribute names can either be identifiers or string literals"
+    )]
+    InvalidAttribute,
+    /// Returned for attempting to use an invalid attribute name in a record name
+    #[error("record literal has invalid attributes")]
+    InvalidAttributesInRecordLiteral,
+    /// Returned for attempting to use an attribute with a namespace
+    #[error("`{0}` cannot be used as an attribute as it contains a namespace")]
+    PathAsAttribute(String),
+    /// Returned when a policy attempts to call a method function-style
+    #[error("`{0}` is a method, not a function. Use a method-style call: `e.{0}(..)`")]
+    FunctionCallOnMethod(crate::ast::Id),
+    /// Returned when the right hand side of a `like` expression is not a constant pattern literal
+    #[error("right hand side of a `like` expression must be a pattern literal. Got: {0}")]
+    InvalidPattern(String),
+    /// Returned when an unexpected node is in the policy scope clause
+    #[error("expected a {expected}, found a `{got}` statement")]
+    WrongNode {
+        /// What the expected AST node kind was
+        expected: &'static str,
+        /// What AST node was present in the policy source
+        got: String,
+    },
+    /// Returned when a policy contains ambiguous ordering of operators.
+    /// This can be resolved by using parenthesis to make order explicit
+    #[error("multiple relational operators (>, ==, in, etc.) without parentheses")]
+    AmbiguousOperators,
+    /// Returned when a policy uses the division operator (`/`), which is not supported
+    #[error("division is not supported")]
+    Division,
+    /// Returned when a policy uses the remainder/modulo operator (`%`), which is not supported
+    #[error("remainder/modulo is not supported")]
+    Modulo,
+    /// Returned when a policy attempts to multiply by a non-constant integer
+    #[error("multiplication must be by an integer literal")]
+    NonConstantMultiplication,
+    /// Returned when a policy contains an integer literal that is out of range
+    #[error(
+        "integer literal `{0}` is too large. Maximum allowed integer literal is `{}`",
+        i64::MAX
+    )]
+    IntegerLiteralTooLarge(u64),
```

Comment:
I don't think this should be `u64`. What gets printed if the input is larger than `u64::MAX`?

Reply:
The only places it gets constructed are places where the CST already has parsed a `u64`. So presumably that error would have to be caught at an earlier level? Changing that might mean changing the CST to have a `i64`, meaning this error can just disappear?